src/main/scala/dev/kamu/cli/commands/PullCommand.scala

```diff
@@ -15,20 +16,35 @@ class PullCommand(
   metadataRepository: MetadataRepository,
   sparkRunner: SparkRunner,
   datasetIDs: Seq[String],
-  all: Boolean
+  all: Boolean,
+  recursive: Boolean
 ) extends Command {
   private val logger = LogManager.getLogger(getClass.getName)
 
   def run(): Unit = {
-    val datasets =
-      if (all)
-        metadataRepository.getAllDatasets()
-      else
-        datasetIDs
-          .map(DatasetID)
-          .map(metadataRepository.getDataset)
-
-    val numUpdated = datasets
+    val dependsOn: DatasetID => List[DatasetID] = {
+      if (all || recursive) {
```

Comment:
this condition doesn't change, so instead of having it in the function it can be move outside of it

Reply:
It seems to me as I assign a function to dependsOn depending on the condition as I would assign a function pointer in other languages. I'll have to check that. Moved the condition outside for now.